dist/tools/kconfiglib/examples/select_specific_implementation_from_module/README.md

```diff
@@ -0,0 +1,74 @@
+# Select a specific choice implementation from a module
+
+This is a design pattern when one must select a specific implementation from
+a module. This is specifically for modules that are not defined in boards or
+applications.
+
+## Reasoning
+
+We need a choice to ensure mutual exclusive options. The issue is that we need
+to detect if multiple modules `REQUIRES_*` different choices. If that happens
```

Comment:
I would add that this modelling is actually not recommended nor desired, and modules should depend on the generic module. One could, however, indicate that certain features in the implementation are 'nice to have' by selecting hidden symbols.